pkg/services/star/api/api.go

```diff
@@ -39,15 +40,25 @@ func (api *API) getDashboardHelper(ctx context.Context, orgID int64, id int64, u
 
 	result, err := api.dashboardService.GetDashboard(ctx, &query)
 	if err != nil {
-		return nil, response.Error(404, "Dashboard not found", err)
+		return nil, response.Error(http.StatusNotFound, "Dashboard not found", err)
 	}
 
 	return result, nil
 }
 
 func (api *API) GetStars(c *contextmodel.ReqContext) response.Response {
+	namespace, identifier := c.SignedInUser.GetNamespacedID()
+	if namespace != identity.NamespaceUser && namespace != identity.NamespaceServiceAccount {
+		return response.Error(http.StatusBadRequest, "Only users and service accounts can star dashboards", nil)
```

Comment:
It's actually pretty weird that service accounts can star dashboards. But oh well...

Reply:
Didn't want to break what existed :sweat_smile: but yeah.